lib/library.rb

```diff
@@ -0,0 +1,40 @@
+class Library
+  attr_reader :name, :books, :authors, :checked_out_books, :times_checked_out
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+    @times_checked_out = {}
+  end
+
+  def add_author(author)
+    @authors.push(author)
+    author.books.each { |book| @books.push(book) }
```

Comment:
good job here; you could also do `@books + author.books` ... Ruby allows you to "concatenate" arrays just by using a `+` operator. :)